src/utils/auth.ts

```diff
@@ -104,6 +107,32 @@ export default class Auth {
     }
   }
 
+  handleCookieBasedAccessTokenAuthentication(
+    accessToken: string,
+    expiresInSeconds: string,
+  ) {
+    // handle any previous location redirects here
+
+    return this.handleNewSession(accessToken, expiresInSeconds)
+  }
+
+  handleNewSession(accessToken: string, expiresInSeconds: string) {
+    return new Promise((resolve, reject) => {
+      this.setSession(accessToken, expiresInSeconds).then(
+        (user) => {
+          identify(user)
+          track('authentication success')
+          resolve(user)
+        },
+        (error) => {
+          console.error(error)
+          this.logout()
+          reject(error)
+        },
+      )
+    })
+  }
```

Comment:
Extracted this bit of session set up logic from `handleAuthentication` so that either auth approach can call it.

Reply:
Also updated it so that `resolve` gets called with the fetched `user` instead of with the oauth `Token` object.